### ISSUE TITLE
ol.source.Vector.refresh() does not update features removed from remote source

### ISSUE THREAD

**Issue by sedot42** [openlayers/openlayers#7044]
Hi all, I dug through closed and open issues but couldn't find an answer, so I'm opening this.

Documentation says about `ol.source.Vector.refresh()`:
> Refreshes the source and finally dispatches a 'change' event.

Imagine having a remote source, for example an API: 			
 ```
 source : new ol.source.Vector({
    format: new ol.format.GeoJSON(),
    url: "/api/tasks"
}
```

After deleting or adding a feature in the database/source, I expected calling `refresh()` to update the state of my map, including removal of deleted features or adding of new ones. However, this does not even trigger a GET request on the source in many cases. On the other side `ol.source.Vector.clear()` works surprisingly well. I expected it to just clear all features from the source but it also fetched up-to-date data from the source. Can anyone reproduce this? Did I misunderstand documentation here? Maybe this also is a chance to improve documentation. `refresh()` was introduced with #4868 which gives some insight, but still does not really explain why this is not working as expected for me.

By the way, thank you very much, all of you, for your work and time spent on this project. 👍

**Comment by Bear4** [comment]
I'm waiting from the beginning for this functionality. Unfortunately till now without result. 
**I hope something going forward :)**

[And solution which isn't good for my needs](http://jsfiddle.net/ahocevar/tLt5wdpz/)
because I have many vector layers which should be refreshed and above solution cause problem when layers switch on/off.

[And my issues, closed](https://github.com/openlayers/openlayers/issues/4558)

So let things rolling  with source.Vector.refresh() 
Thanks for OL.

**Comment by ahocevar** [comment]
I think the confusion here is that for sources configured with a `url` and withoug `strategy`, `ol.source.Vector#clear()` does what people expect `#refresh()` to do. But the latter simply dispatches a `'change'` event, nothing else.

A pull request could change the behavior of both methods, so that `#clear()` does not reload features, and `#refresh` always reloads features, even when the source was configured with a `strategy`.

**Comment by sedot42** [comment]
I finally had a look at the source, because I want to fix this. I think `ol.source.Vector#clear()` doesn't need to be changed, it does exactly what one would expect. However, I don't exactly understand why it triggers a refresh. If the source was refreshed because `#clear()` emits a `'change'` event, then `#refresh()` should trigger a refresh as well since it emits a `'change'` event as well. 
@ahocevar Can you elaborate, please? 

<details>
 <summary>source of `ol.source.Vector#clear()`</summary>

```js
_ol_source_Vector_.prototype.clear = function(opt_fast) {
  if (opt_fast) {
    for (var featureId in this.featureChangeKeys_) {
      var keys = this.featureChangeKeys_[featureId];
      keys.forEach(_ol_events_.unlistenByKey);
    }
    if (!this.featuresCollection_) {
      this.featureChangeKeys_ = {};
      this.idIndex_ = {};
      this.undefIdIndex_ = {};
    }
  } else {
    if (this.featuresRtree_) {
      this.featuresRtree_.forEach(this.removeFeatureInternal, this);
      for (var id in this.nullGeometryFeatures_) {
        this.removeFeatureInternal(this.nullGeometryFeatures_[id]);
      }
    }
  }
  if (this.featuresCollection_) {
    this.featuresCollection_.clear();
  }

  if (this.featuresRtree_) {
    this.featuresRtree_.clear();
  }
  this.loadedExtentsRtree_.clear();
  this.nullGeometryFeatures_ = {};

  var clearEvent = new _ol_source_Vector_.Event(_ol_source_VectorEventType_.CLEAR);
  this.dispatchEvent(clearEvent);
  this.changed();
};
```
</details>

<details>
 <summary>source of `ol.source.Source#refresh()`</summary>

```js
/**
 * Refreshes the source and finally dispatches a 'change' event.
 * @api
 */
_ol_source_Source_.prototype.refresh = function() {
  this.changed();
};
```
</details>

**Comment by simonseyock** [comment]
The loading is done by the loadFeatures method (https://github.com/openlayers/openlayers/blob/4275a66b496fe13d0281c19326fd2c0b72dfec19/src/ol/source/Vector.js#L740). Which is called by the renderer every time the source dispatches a change event.

And the clear method also clears the loadedExtentsRtree, which is checked in loadFeatures.






Why it is reloaded is because of the line

**Comment by sedot42** [comment]
Ok. Would overriding `ol.source.Vector#refresh()` with a call to `#clear()` solve the confusion while retaining functionality for all other sources?
For example:
```js
_ol_source_Vector_.prototype.refresh = function() {
    this.clear()
}
```

**Comment by yaugenka** [comment]
imho, `clear()` is the oposite of `load()` (the latter does not exist) and both make sense for clearing/loading features from/into a source which parent layer is not added to a map. If a source is added to a map, then only refresh() makes sense because it combines the two actions.

**Comment by sedot42** [comment]
Should be fixed by #9250
